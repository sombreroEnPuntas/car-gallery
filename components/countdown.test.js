// @flow
import React from 'react'
import { shallow } from 'enzyme'

// Tested unit
import TestedComponent from './countdown'

// Mocks
jest.useFakeTimers()

const getProps = customProps => ({
  ...customProps,
})

describe('Countdown', () => {
  const wrapper = shallow(<TestedComponent {...getProps()} />)

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('decreases the number', () => {
    jest.runOnlyPendingTimers()
    expect(wrapper.state().time).toBe(2)
    jest.runOnlyPendingTimers()
    expect(wrapper.state().time).toBe(1)
  })

  it('unmounts cleanly', () => {
    wrapper.unmount()
    // If an Interval is still existing, jest will keep running timers until
    // an infinite recursion error ocurrs
    jest.runAllTimers()
  })
})
