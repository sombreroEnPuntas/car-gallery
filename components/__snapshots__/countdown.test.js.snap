// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown matches the snapshot 1`] = `
<p>
  Retry in: 3
</p>
`;
